File: scripts/test_solutions/424_longest_repeating_char_replacement.js
var characterReplacement = function(s, k) {
    const count = {};
    let left = 0, maxCount = 0, maxLen = 0;
    for (let right = 0; right < s.length; right++) {
        count[s[right]] = (count[s[right]] || 0) + 1;
        maxCount = Math.max(maxCount, count[s[right]]);
        while (right - left + 1 - maxCount > k) { count[s[left]]--; left++; }
        maxLen = Math.max(maxLen, right - left + 1);
    }
    return maxLen;
};

File: scripts/test_solutions/424_longest_repeating_char_replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        unordered_map<char, int> count;
        int left = 0, maxCount = 0, maxLen = 0;

        for (int right = 0; right < s.length(); right++) {
            count[s[right]]++;
            maxCount = max(maxCount, count[s[right]]);

            while (right - left + 1 - maxCount > k) {
                count[s[left]]--;
                left++;
            }
            maxLen = max(maxLen, right - left + 1);
        }
        return maxLen;
    }
};
